### PR TITLE
replace synchronization primitives with those from parking_lot

### LINKED ISSUE #1374: Replace locks with parking_lot alternatives
Could be faster, might not be. Definitely needs profiling. These mutexes are supposed to be more space and time efficient, as well as consistent across platforms.

### PR DESCRIPTION
**Pull request by rphmeier**
closes #1374

### CODE REVIEW

File: dapps/src/lib.rs [hunk 1]
```diff
@@ -153,7 +153,7 @@ impl Server {
 
 	/// Set callback for panics.
 	pub fn set_panic_handler<F>(&self, handler: F) where F : Fn() -> () + Send + 'static {
-		*self.panic_handler.locked() = Some(Box::new(handler));
+		*self.panic_handler.lock().unwrap() = Some(Box::new(handler));
```
Comment (arkpar) [inline dapps/src/lib.rs]: why not keep using `Lockable` here?

**Comment by rphmeier** [comment]
will make this in-progress until i get some benchmark data

**Comment by gavofyork** [comment]
Results in a ~5-10% speedup for syncing the first 200 blocks. code is cleaner, too.

File: dapps/src/lib.rs (hunk 1 above)
Comment (rphmeier) [inline dapps/src/lib.rs]: reverted to `std` primitives here because the jsonrpc server requires them for its panic handler

**Comment by rphmeier** [comment]
gives a ~3% speedup on my machine importing the first 250k blocks.

**Pull request merged**